Issue #11 - Enhance the Issuer section to allow clicking through to issuing certificate tab

MauAraujo opened the pull request:
I added the expected functionality as observed in the image below: 
![issuers](https://user-images.githubusercontent.com/19579265/53773946-15a8de00-3eb2-11e9-8979-c68dada6cc5b.gif)
@april @glawrence, I would like some feedback on the styling and general implementation. 
Maybe in the future some animations could be added to make it friendlier to the user. Thanks in advance.

glawrence commented:
Thanks @MauAraujo that looks great and is exactly what I had in mind, nice job!

The only minor comment would be whether you could turn the link off for Issuer Section in the Root CA tab, of course you will need to detect it is a built-in root but that is already being done as the tab title has "(built-in root)" on the end.

april commented:
I agree with @glawrence about removing the link on the root CA (since it is self-signed).

As for the styles, I'm not 100% certain since that's a bit outside of my wheelhouse. I'm fairly sure that it shouldn't have a different "visited" color, and it's a bit abrupt that it's transitioning tabs. Let me see if I can get someone from the Firefox UX team to comment on this.

MauAraujo commented:
@april @glawrence I removed the link on the root CA, and I also fixed the style so that the link is always the same color, even when it has been previously visited. I agree that the transitioning is rather abrupt. If it is OK, I can start looking into adding some light animations to make it look smoother and friendlier to the user. Thanks in advance for your feedback.
![noroot](https://user-images.githubusercontent.com/19579265/53831915-923bcb00-3f4b-11e9-9561-4a22f6ef71df.gif)

glawrence commented:
This looks great now: keeping the links blue and not making the root CA issuer a link

However, it looks like you have pulled in some code from #42 which is a little confusing, probably better to keep these separate

MauAraujo commented:
Sorry, that wasn't supposed to be there, it was a mistake. I fixed it now to only show the relevant code.

MauAraujo commented:
After testing the code again, I noticed that when a certificate had more than 3 tabs, the tabbing was not working properly. I fixed it and now it works properly when there are multiple tabs. 
![multipletabs](https://user-images.githubusercontent.com/19579265/53897742-62003500-3ffc-11e9-87d1-27b91a9a5d1d.gif)

glawrence commented:
I can confirm the latest code works in terms of the Root CA and the 4 tabs that show at https://www.hs-bremerhaven.de/start/ also work, so this is looking great, nice work! @april I think you could merge this, unless you want some tab change animation, but that could go in a new issue, maybe?

april commented:
Hmm, I'm using the currently released public version of Certainly Something and it works fine with four tiers of certificates. Can you help me understand what you mean about it not working with four tabs?

MauAraujo commented:
Yes, sorry if I wasn't clear before. After I reviewed my changes, I discovered that the links for switching tabs between issuers did not work properly when the certificate had more than 3 issuer tabs. This was because previously I hadn't considered the case, but I fixed my code in the last commit so that it works regardless of the number of issuer tabs.

april commented on src/viewer/js/render.js:
```diff
@@ -24,6 +24,7 @@ const postRender = () => {
   const buttons = document.getElementById('certificates').getElementsByClassName('panel-section-tabs-button');
   const certificates = document.getElementsByClassName('certificate');
   const longhexes = document.getElementsByClassName('long-hex');
+  const issuerlink = document.getElementsByClassName('issuer-link');
```
This should probably be pluralized, since there is more than one.

april commented on src/viewer/index.handlebars:
```diff
@@ -43,7 +43,7 @@
           <div class="panel-section" id="panel-section-tabs">
             {{#each certs}}
-            <div class="panel-section-tabs-button {{#if @first}}selected{{/if}}" data-certificate-index="{{@index}}">{{ this.subject.cn }} {{#if this.isBuiltInRoot}}(built-in root){{/if}}</div>
+              <div class="panel-section-tabs-button {{#if @first}}selected{{/if}}" data-certificate-index="{{@index}}">{{ this.subject.cn }} {{#if this.isBuiltInRoot}}(built-in root){{/if}}</div>
```
Why was this line changed?

MauAraujo replied:
Sorry, adding that white space was unnecessary.

april commented on src/viewer/index.handlebars:
```diff
@@ -69,8 +69,16 @@
-            <div class="text">{{lookup this 0}}</div>
-            <div class="text">{{lookup this 1}}</div>
+              <div class="text">{{lookup this 0}}</div>
+              {{#if (eq (lookup this 0)  "Common Name")}}
```
You probably want to link the last entry, since Common Name will eventually be localized.

april commented on src/viewer/js/render.js:
```diff
@@ -47,6 +48,18 @@ const postRender = () => {
   }
 
+    // setup event handler for issuer navigation
+    for (let node of issuerlink) {
```
This seems very complicated. The buttons are already numbered (via `data-certificate-index`), and it seems as if it would be better to find the current index and then click the button that has an index that is +1.

glawrence commented:
I have tested the latest changes and it works for me, so that's a "Test: Passed" from me

glawrence commented:
Sorry @april but how do you feel about merging this one, I can't see anything outstanding

MauAraujo commented:
Is there anything that is missing? Should I proceed with adding animations? I honestly thought that was better as a separate issue, so I was waiting on this one to be merged. Please advise on what to so next.

glawrence commented:
I think animation should be a separate issue, allows this to be closed off and also means another issue/PR can discuss just animation
